src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -508,10 +508,10 @@ public ContextualReflectionScope EnterContextualReflection()
         /// Returns a disposable ContextualReflectionScope for use in a using block. When the using calls the
         /// Dispose() method, it restores the ContextualReflectionScope to its previous value.
         /// </remarks>
-        public static ContextualReflectionScope EnterContextualReflection(Assembly activating)
+        public static ContextualReflectionScope EnterContextualReflection(Assembly? activating)
         {
             return activating != null ?
-                GetLoadContext(activating).EnterContextualReflection() :
+                GetLoadContext(activating)?.EnterContextualReflection() ?? default :
```

Comment:
This is changing the implementation.  Is there actually a risk of a null dereference here?

Reply:
Yes... `GetLoadContext` can actually return null if `activating` is not a `RuntimeAssembly` https://github.com/dotnet/coreclr/blob/206f9680303134731862a35338618f2cbd225651/src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.CoreCLR.cs#L254

That is why I thought of guarding it, but actually , you're right, we shouldn't change the implementaiton, just maybe open an issue since that is the same case now, it could cause a null reference exception, especially since it is a public API and any implementation of `Assembly` could be passed down and just return `null`... I could add a TODO comment and open an issue to discuss this further.

Reply:
> just maybe open an issue since that is the same case now

Let's do that.  Please open an issue, `!` away the warning here, and tag this with `// TODO: https://issue/You/Opened`.  Thanks.

Reply:
Sounds good. Thanks for taking a look!